base/cvd/cuttlefish/host/commands/cvd/group_selector.cpp

```diff
@@ -43,4 +51,84 @@ std::string GroupSelector::Menu() {
   return ss.str();
 }
 
+Result<selector::LocalInstanceGroup> PromptUserForGroup(
+    InstanceManager& instance_manager, const RequestWithStdio& request,
+    const cvd_common::Envs& envs, const cvd_common::Args& selector_args) {
+  // show the menu and let the user choose
+  std::vector<selector::LocalInstanceGroup> groups =
+      CF_EXPECT(instance_manager.FindGroups(selector::Queries{}));
+  groups.erase(std::remove_if(groups.begin(), groups.end(),
+                              [](const auto& group) {
+                                return !group.HasActiveInstances();
+                              }),
+               groups.end());
+  auto menu = SelectionMenu(groups);
+
+  CF_EXPECT_EQ(WriteAll(request.Out(), menu + "\n"), (ssize_t)menu.size() + 1);
+  std::unique_ptr<InterruptibleTerminal> terminal_ =
+      std::make_unique<InterruptibleTerminal>(request.In());
+
+  const bool is_tty = request.Err()->IsOpen() && request.Err()->IsATTY();
+  while (true) {
+    std::string question = "";
+    CF_EXPECT_EQ(WriteAll(request.Out(), question), (ssize_t)question.size());
+
+    std::string input_line = CF_EXPECT(terminal_->ReadLine());
+    int selection = -1;
+    std::string chosen_group_name;
+    if (android::base::ParseInt(input_line, &selection)) {
+      const int n_groups = groups.size();
+      if (n_groups <= selection || selection < 0) {
+        std::string out_of_range = fmt::format(
```

Comment:
nit: the function was an easy read until seeing the `TerminalColor(is_tty, TerminalColors::kBoldRed) lines.  In favor of readability wondering a preprocessor directives might be justifiable here, it's ok if not.

Reply:
I think this can be improved without resorting to preprocessor directives. Will do in follow up PR.